lib/vmpooler/pool_manager.rb

```diff
@@ -448,39 +460,111 @@ def _check_snapshot_queue
       unless vm.nil?
         begin
           vm_name, snapshot_name = vm.split(':')
-          revert_vm_snapshot(vm_name, snapshot_name)
+          revert_vm_snapshot(vm_name, snapshot_name, vsphere)
         rescue
           $logger.log('s', "[!] [snapshot_manager] snapshot revert appears to have failed")
         end
       end
     end
 
+    def find_vsphere_pool_vm(pool, vm, vsphere)
+      vsphere.find_vm(vm) || vsphere.find_vm_heavy(vm)[vm]
+    end
+
+    def migration_limit(migration_limit)
+      # Returns migration_limit setting when enabled
+      return false if migration_limit == 0 or not migration_limit
+      migration_limit if migration_limit >= 1
+    end
+
+    def migrate_vm(vm, pool, vsphere)
+      Thread.new do
+        _migrate_vm(vm, pool, vsphere)
+      end
+    end
+
+    def _migrate_vm(vm, pool, vsphere)
+      begin
+        $redis.srem('vmpooler__migrating__' + pool, vm)
+        vm_object = find_vsphere_pool_vm(pool, vm, vsphere)
+        parent_host, parent_host_name = get_vm_host_info(vm_object)
+        migration_limit = migration_limit $config[:config]['migration_limit']
+
+        if not migration_limit
+          $logger.log('s', "[ ] [#{pool}] '#{vm}' is running on #{parent_host_name}")
+        else
+          migration_count = $redis.smembers('vmpooler__migration').size
```

Comment:
Wondering if `scard` would work here instead of the `smembers().size`.  Also, we might be able to fetch this at the top instead of inline, but I may not understand the intricacies here.